node/pkg/txverifier/evm.go

```diff
@@ -639,43 +643,56 @@ func parseLogMessagePublishedPayload(
 	return &TransferDetails{
 		PayloadType:      VAAPayloadType(hdr.Type),
 		AmountRaw:        hdr.Amount,
-		OriginAddressRaw: hdr.OriginAddress.Bytes(),
+		OriginAddressRaw: hdr.OriginAddress,
 		TokenChain:       vaa.ChainID(hdr.OriginChain),
 		TargetAddress:    hdr.TargetAddress,
 		// these fields are populated by RPC calls later
 		Amount:        nil,
-		OriginAddress: common.Address{},
+		OriginAddress: vaa.Address{},
 	}, nil
 }
 
 // fetchLogMessageDetails makes requests to the token bridge and token contract to get detailed, wormhole-specific information about
 // the transfer details parsed from a LogMessagePublished event.
-func (tv *TransferVerifier[ethClient, connector]) fetchLogMessageDetails(details *TransferDetails) (newDetails *TransferDetails, decimalErr error) {
-	// This function adds information to a TransferDetails struct, filling out its uninitialized fields.
-	// It populates the following fields:
-	// - Amount: populate the Amount field by denormalizing details.AmountRaw.
-	// - OriginAddress: use the wormhole ChainID and OriginAddressRaw to determine whether the token is wrapped.
-
-	// If the token was minted on the chain monitored by this program, set its OriginAddress equal to OriginAddressRaw.
-	var originAddress common.Address
+// Returns nil on error.
+func (tv *TransferVerifier[ethClient, connector]) fetchLogMessageDetails(details *TransferDetails) (*TransferDetails, error) {
+	// Populate the Amount, OriginAddress, and token decimals for assets
+	// recorded in LogMessagePublished events for this receipt.
+	// The Amount and OriginAddress are encoded as raw bytes in the log but they need to be processed in
+	// order to verify transfers:
+	// - AmountRaw must be (de)normalized according to its decimals
+	// - OriginAddressRaw use the wormhole chain ID and address to determine whether the token is wrapped.
+	var (
+		originAddress vaa.Address
+		newDetails    = *details
+		decimals      uint8
+		// The token address that will be queried for decimal information.
+		queryAddr common.Address
+	)
+	// At this point, details.TokenChain should be set to the chain ID in the token transfer's payload header.
+	// TODO: I'm not sure that this is correct. The address should be validated either way.
 	if details.TokenChain == tv.chainIds.wormholeChainId {
```

Comment:
I'm confused about the necessity of this code. Here's my understanding of it currently: 
- If the TokenChain is the same as the current chain, the origin address is set to the current `OriginAddressRaw` from the payload. That makes sense. 
- In the case that the TokenChain is *different* than the current chain, we do some other logic:
    - Query the OriginAddressRaw and the TokenChain with `wrappedAddr` to fetch its locally deployed token. Again, that makes sense. 
    - Call `nativeContract` with the address from the previous step - the token implementation. 
    - Set the result of the `nativeContact` call to the `originAddress`. 

What I'm confused about is the circular usage in the non-native case. The result of the nativeContract is the `OriginAddressRaw`, right?

Reply:
Hmmmm yeah I think you're right, I don't remember why this was written this way. I'll try deleting it and see if something jumps out.

Reply:
Okay, I'm glad I'm not crazy. Thanks for considering.

Reply:
This part of the code is totally different in the latest commit.